### ISSUE TITLE
Persistent attribute instance even after successful deletion 

### ISSUE THREAD

**Issue by cpmlreef** [vaticle/typedb#6802]
Please replace every line in curly brackets { like this } with an appropriate answer, and remove this line.

## Description

I am trying basic operations in Studio and via python client to delete an attribute instance with this query (including commit): 
Running> Delete query:
match
$w isa website,
    has url "https://apple.news/",
    has description $d,
    has last_crawl_date $lcd;
delete
$lcd isa last_crawl_date;

Result> Delete query successfully deleted things from the database.

Then, to check I run: 
match $w isa website, has url "https://apple.news/", has title $t, has last_crawl_date $lcd;
get $w, $t, $lcd;

To get: 
Result> Match query successfully matched concepts in the database:
{
    $t "Apple News" isa title;
    $w iid 0x826e8007800000000000001a isa website;
    $lcd 2023-05-02T15:55:09 isa last_crawl_date;
}
Completed

**The last_crawl_date instance is still found.** 

## Environment

1. OS (where TypeDB server runs): { Windows 10 }
2. TypeDB version (and platform): { TypeDB 2.16.1 }
3. TypeDB client: { client-python & Studio 2.14.2}
4. Other environment details:

## Reproducible Steps

Steps to create the smallest reproducible scenario:
1. { Open session and run query: match $w isa website, has url "https://apple.news", has description $d, has last_crawl_date $lcd;
delete $lcd isa last_crawl_date; }
2. { After successful message, manual commit in Studio / automatic commit via python client. Session closed. }
3. { Run query in Studio: match $w isa website, has url "https://apple.news", has title $t, has last_crawl_date $lcd;
get $w, $t, $lcd; }
4. { See persistent attribute instance  }

## Expected Output

The attribute instance should be deleted and the query "match $w isa website, has url "https://apple.news", has title $t, has last_crawl_date $lcd;
get $w, $t, $lcd; "
should result in no match found.

## Actual Output

Result> Match query successfully matched concepts in the database:
{
    $t "Apple News" isa title;
    $w iid 0x826e8007800000000000001a isa website;
    **$lcd 2023-05-02T15:55:09 isa last_crawl_date**;
}
Completed
 
## Additional information

{ Any additional information, including logs or screenshots if you have any. }

**Comment by cpmlreef** [comment]
So I could replicate the behaviour and could find the error. Although it should (theoretically) through an exception. 

The initial query included the attribute "description d", which was empty for the specific website: 
Running> Delete query:
match
$w isa website,
has url "https://apple.news/",
has description $d,
has last_crawl_date $lcd;
delete
$lcd isa last_crawl_date;

After removing the attr. description from the query, the deletion was successful (wich is good). 

Nevertheless, the match query should have resultet in no match found, as I was trying to find an entity with an empty attribute and thus, the logical result would have been a non-deletion (which it was) but with a proper error message.

**Comment by shiladitya-mukherjee** [comment]
Your first match query (when you included the description variable inside the match query) did not actually match any datapoints in the database. Your confusion resulted from us printing `## Result> Delete query successfully deleted things from the database.` regardless of the cardinality of the set of deleted objects (so we print this even when there are 0 deleted objects). 

If you run the first match query without the delete, you will see that we indeed match nothing in the database.

Closing this issue.
